### PR TITLE
Vue Project Setup

### PR DESCRIPTION
**Pull request by paglias**
WIP

### CODE REVIEW

**Comment by paglias** [comment]
What's been done
- [x] old client tests files move to /test/client-old (in develop)
- [x] gulp tasks moved to /gulp
- [x] webpack files added to /webpack
- [x] client files added to /website/client
- [x] client tests added to /test/client
- [x] client scripts added to package.json with `client:` prefix

What's missing:
- [x] Move /gulp and /webpack in one folder? `/build-process` maybe?
- [x] Babel needs to be split in server version + client (what about tests & common)
- [x] Webpack paths needs to be updated
- [x] re-enable eslint as a preLoader in webpack
- [x] fix static assets not being copied into /dist
- [x] Eslint needs to be added for new client files and tests
- [x] client side tests are not working (and need to be linted, for now they're in .eslintignore)
- [x] deps for tests are probably conflicting with ours (karma, chai, sinon, phantomjs, ...)
- [ ] do we need `babel-preset-stage-2` for the client?

```
    "babel-preset-stage-2": "^6.0.0",

    "karma": "^0.13.15",
    "karma-coverage": "^0.5.5",
    "karma-mocha": "^0.2.2",
    "karma-phantomjs-launcher": "^1.0.0",
    "karma-sinon-chai": "^1.2.0",
    "karma-sourcemap-loader": "^0.3.7",
    "karma-spec-reporter": "0.0.24",
    "karma-webpack": "^1.7.0",
    "mocha": "^2.4.5",
    "chai": "^3.5.0",
    "sinon": "^1.17.3",
    "sinon-chai": "^2.8.0",
    "isparta-loader": "^2.0.0",
    "phantomjs-prebuilt": "^2.1.3",
    "chromedriver": "^2.21.2",
    "nightwatch": "^0.8.18",
    "selenium-server": "2.53.0",
```

**Comment by paglias** [comment]
Building the webpack project now works for both dev and prod environments

**Comment by paglias** [comment]
Remaining depencencies that needs to be evaluated:

> babel-preset-stage-2@^6.0.0 karma-phantomjs-launcher@^1.0.0 karma-spec-reporter@0.0.24 phantomjs-prebuilt@^2.1.3

To run the tests install these deps and run `npm run client:test`

> npm install babel-preset-stage-2@^6.0.0 karma-phantomjs-launcher@^1.0.0 karma-spec-reporter@0.0.24 phantomjs-prebuilt@^2.1.3

**Comment by paglias** [comment]
I'd like to merge if no one has objections

**Review by crookedneighbor — APPROVED** [review]

File: test/client/unit/index.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// Polyfill fn.bind() for PhantomJS
+/* eslint-disable no-extend-native */
+Function.prototype.bind = require('function-bind');
```
Comment (crookedneighbor) [inline test/client/unit/index.js]: This shouldn't be necessary with phantom 2

Comment (paglias) [reply test/client/unit/index.js]: Came with the template, I'll remove it

**Pull request merged**